### PR TITLE
[Implementation]: Android - Send a custom test header on search request

### PR DESCRIPTION
**Pull request by anikiki**
Task/Issue URL: https://app.asana.com/0/1200581511062568/1208723599553913/f

### Description
Added a custom test header.

### Steps to test this PR
See **Test scenarios** section in the description of the Asana task -> https://app.asana.com/0/1200581511062568/1208723599553913/f

### CODE REVIEW

**Comment by anikiki** [comment]
* **#5266** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/5266?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `develop`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @anikiki and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: app/src/main/java/com/duckduckgo/app/browser/AndroidFeaturesHeader.kt [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+import com.duckduckgo.anvil.annotations.ContributesRemoteFeature
+import com.duckduckgo.common.utils.plugins.headers.CustomHeadersPlugin
+import com.duckduckgo.di.scopes.AppScope
+import com.duckduckgo.feature.toggles.api.Toggle
+import com.squareup.anvil.annotations.ContributesMultibinding
+import javax.inject.Inject
+
+@ContributesMultibinding(scope = AppScope::class)
+class AndroidFeaturesHeaderPlugin @Inject constructor(
```
Comment (malmstein) [inline app/src/main/java/com/duckduckgo/app/browser/AndroidFeaturesHeader.kt:27]: nit: rename to `AndroidBrowserFeaturesHeaderPlugin`

File: app/src/main/java/com/duckduckgo/app/browser/BrowserTabViewModel.kt [hunk 2]
```diff
@@ -1073,8 +1073,12 @@ class BrowserTabViewModel @Inject constructor(
     }
 
     private fun getUrlHeaders(url: String?): Map<String, String> {
-        url?.let {
-            return gpc.getHeaders(url)
+        url?.let { requestUrl ->
+            val customHeaders = mutableMapOf<String, String>()
+            customHeadersPlugin.getPlugins().forEach {
```
Comment (malmstein) [inline app/src/main/java/com/duckduckgo/app/browser/BrowserTabViewModel.kt:1078]: should this be in a customHeadersProvider? All logic can be encapsulated there.

**Review by malmstein — CHANGES_REQUESTED** [review]
We need to capture the headers in `BrowserWebViewClient` and add them in `shouldOverrideRequest`. Returning true after we call `loadUrl` with the added headers.

File: app/src/main/java/com/duckduckgo/app/browser/AndroidFeaturesHeader.kt (hunk 1 above)
Comment (anikiki) [reply app/src/main/java/com/duckduckgo/app/browser/AndroidFeaturesHeader.kt:26]: Didn't change this.

File: app/src/main/java/com/duckduckgo/app/browser/BrowserTabViewModel.kt (hunk 2 above)
Comment (anikiki) [reply app/src/main/java/com/duckduckgo/app/browser/BrowserTabViewModel.kt]: Updated, thanks :+1:

**Comment by anikiki** [comment]
> We need to capture the headers in `BrowserWebViewClient` and add them in `shouldOverrideRequest`. Returning true after we call `loadUrl` with the added headers.

Good catch, thanks :+1:  Updated.

**Pull request merged**